controllers/users_config_map_controller.go

```diff
@@ -0,0 +1,303 @@
+			log.Info("Config map with additional users doesn't have the key", "ConfigMapName", additionalUsersConfigMapName, "ConfigMapKey", additionalUsersConfigMapKey, l.Action, l.ActionView)
+			removeTeam = true
+		} else {
+			additionalUsers = strings.Fields(strings.TrimSpace(additionalUsersStr))
+			log.Info("Additional users configured in config map", "AdditionalUsers", additionalUsers)
+		}
+	}
+
+	// reread quay token
+	r.QuayClient = r.BuildQuayClient(log)
+
+	// remove team if config map was removed, or doesn't contain any users
+	if removeTeam {
+		log.Info("Will remove team", "TeamName", teamName)
+		if err := r.QuayClient.DeleteTeam(r.QuayOrganization, teamName); err != nil {
+			log.Error(err, "failed to remove team", "TeamName", teamName, l.Action, l.ActionDelete)
+			return ctrl.Result{}, err
+		}
+
+		// remove finalizer after team is removed
+		if !configMap.DeletionTimestamp.IsZero() {
+			if controllerutil.ContainsFinalizer(configMap, ConfigMapFinalizer) {
+				controllerutil.RemoveFinalizer(configMap, ConfigMapFinalizer)
+				if err := r.Client.Update(ctx, configMap); err != nil {
+					log.Error(err, "failed to remove config map finalizer", "ConfigMapName", additionalUsersConfigMapName, "Finalizer", ConfigMapFinalizer, l.Action, l.ActionUpdate)
+					return ctrl.Result{}, err
+				}
+				log.Info("finalizer removed from config map", l.Action, l.ActionDelete)
+			}
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	// fetch image repositories in the namespace
+	imageRepositoryList := &imagerepositoryv1alpha1.ImageRepositoryList{}
+	if err := r.Client.List(ctx, imageRepositoryList, &client.ListOptions{Namespace: configMap.Namespace}); err != nil {
+		log.Error(err, "failed to list ImageRepositories", l.Action, l.ActionView)
+		return ctrl.Result{}, err
+	}
+
+	// get image repositories names
+	repositoriesToAdd := []string{}
+	for _, repository := range imageRepositoryList.Items {
+		repositoriesToAdd = append(repositoriesToAdd, repository.Spec.Image.Name)
+	}
+
+	// get team members, if team doesn't exist it will be created
+	log.Info("Ensure team", "TeamName", teamName)
+	teamMembers, err := r.QuayClient.EnsureTeam(r.QuayOrganization, teamName)
+	if err != nil {
+		log.Error(err, "failed to get team members", "TeamName", teamName, l.Action, l.ActionView)
+		return ctrl.Result{}, err
+	}
+
+	// get team permissions
+	teamPermissions, err := r.QuayClient.ListRepositoryPermissionsForTeam(r.QuayOrganization, teamName)
+	if err != nil {
+		log.Error(err, "failed to get team permissions", "TeamName", teamName, l.Action, l.ActionView)
+		return ctrl.Result{}, err
+	}
+	// get repositories for which team has permissions
+	reposTeamPermissions := []string{}
+	for _, repoPermission := range teamPermissions {
+		reposTeamPermissions = append(reposTeamPermissions, repoPermission.Repository.Name)
+	}
+	log.Info("Team has repository permissions", "TeamName", teamName, "Repositories", reposTeamPermissions)
+
+	// grant repo permissions to the team
```

Comment:
Can we have strategy similar to users: list of repositories to process?

Reply:
using new generic function as well